# A Third-party Analysis of the AASL Standards

Jinren Yuan

This section contains the analysis of **collaborators**.

In [1]:
import pandas as pd

In [2]:
df_c = pd.read_csv("data/AASL Standards Survey - Collaborators - Sheet.csv")
df_c.shape

(43, 149)

In [3]:
df_c.head()

Respondent ID  Collector ID           Start Date             End Date  \
0            NaN           NaN                  NaN                  NaN   
1   1.184746e+11   453683477.0  2023-11-20 10:06:21  2023-11-20 10:06:31   
2   1.184745e+11   453683477.0  2023-11-20 07:45:34  2023-11-20 07:45:52   
3   1.184742e+11   453683477.0  2023-11-19 23:24:27  2023-11-19 23:57:41   
4   1.184732e+11   453683477.0  2023-11-17 19:52:30  2023-11-17 19:54:32   

   IP Address  Email Address  First Name  Last Name  Custom Data 1  \
0         NaN            NaN         NaN        NaN            NaN   
1         NaN            NaN         NaN        NaN            NaN   
2         NaN            NaN         NaN        NaN            NaN   
3         NaN            NaN         NaN        NaN            NaN   
4         NaN            NaN         NaN        NaN            NaN   

  Which of the following positions most closely reflects your job role? Please select all that apply.  \
0                                  None of the above                                                    
1                                  None of the above                                                    
2                                  None of the above                                                    
3                                                NaN                                                    
4                                                NaN                                                    

   ...              Unnamed: 139                       Unnamed: 140  \
0  ...  English language version  Another language (please specify)   
1  ...                       NaN                                NaN   
2  ...                       NaN                                NaN   
3  ...                       NaN                                NaN   
4  ...                       NaN                                NaN   

  What is the primary language spoken in your work environment?  \
0                                           Response              
1                                                NaN              
2                                                NaN              
3                                                NaN              
4                                                NaN              

             Unnamed: 142  \
0  Other (please specify)   
1                     NaN   
2                     NaN   
3                     NaN   
4                     NaN   

  Are you of Hispanic or Latino background – such as Mexican, Puerto Rican, Cuban, or another Latin American background?  \
0                                           Response                                                                       
1                                                NaN                                                                       
2                                                NaN                                                                       
3                                                NaN                                                                       
4                                                NaN                                                                       

  Which of the following categories best describes your race or ethnicity?  \
0                                           Response                         
1                                                NaN                         
2                                                NaN                         
3                                                NaN                         
4                                                NaN                         

             Unnamed: 145  \
0  Other (please specify)   
1                     NaN   
2                     NaN   
3                     NaN   
4                     NaN   

  Which of the following categories best describes your employment status?  \
0                     

In [4]:
df_c.columns

Index(['Respondent ID', 'Collector ID', 'Start Date', 'End Date', 'IP Address',
       'Email Address', 'First Name', 'Last Name', 'Custom Data 1',
       'Which of the following positions most closely reflects your job role? Please select all that apply.',
       ...
       'Unnamed: 139', 'Unnamed: 140',
       'What is the primary language spoken in your work environment?',
       'Unnamed: 142',
       'Are you of Hispanic or Latino background – such as Mexican, Puerto Rican, Cuban, or another Latin American background?',
       'Which of the following categories best describes your race or ethnicity?',
       'Unnamed: 145',
       'Which of the following categories best describes your employment status?',
       'Unnamed: 147', 'Is there anything else you would like to add?'],
      dtype='object', length=149)

In [5]:
# drop irrelavent ID and date fields
df_c = df_c.drop(columns=['Respondent ID', 'Collector ID', 'Start Date', 'End Date'])

# drop fields that contain only missing values (NaN)
df_c = df_c.dropna(axis=1, how='all')
df_c.head()

Which of the following positions most closely reflects your job role? Please select all that apply.  \
0                                  None of the above                                                    
1                                  None of the above                                                    
2                                  None of the above                                                    
3                                                NaN                                                    
4                                                NaN                                                    

                          Unnamed: 10                          Unnamed: 11  \
0  Teacher (e.g. educational faculty)  Support staff (e.g. teacher’s aide)   
1                                 NaN                                  NaN   
2                                 NaN                                  NaN   
3  Teacher (e.g. educational faculty)                                  NaN   
4  Teacher (e.g. educational faculty)                                  NaN   

                                         Unnamed: 12  \
0  Building Level Technology (e.g. tech specialis...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                        Unnamed: 13  \
0  Instructional Facilitator/ Coach   
1                               NaN   
2                               NaN   
3  Instructional Facilitator/ Coach   
4                               NaN   

  Which of the following best describes your work environment? Please select all that apply. (i.e. If you are in a combined setting you may select multiple levels.)  \
0                                  Elementary school                                                                                                                   
1                                                NaN                                                                                                                   
2                                                NaN                                                                                                                   
3                                                NaN                                                                                                                   
4                                  Elementary school                                                                                                                   

                           Unnamed: 15  Unnamed: 16         Unnamed: 17  \
0  Middle school or junior high school  High school  Multiple buildings   
1                                  NaN          NaN                 NaN   
2                                  NaN          NaN                 NaN   
3  Middle school or junior high school  High school                 NaN   
4                                  NaN          NaN                 NaN   

              Unnamed: 18  ...              Unnamed: 139  \
0  School District office  ...  English language version   
1                     NaN  ...                       NaN   
2                     NaN  ...                       NaN   
3                     NaN  ...                       NaN   
4                     NaN  ...                       NaN   

                        Unnamed: 140  \
0  Another language (please specify)   
1                                NaN   
2                                NaN   
3                                NaN   
4                                NaN   

  What is the primary language spoken in your work environment?  \
0                                           Response              
1                                                NaN              
2                                                NaN              


In [6]:
# Drop Q2 - Q4
df_c = df_c.drop(df_c.columns[5:32], axis=1)

In [7]:
df_c.shape

(43, 113)

In [8]:
# Drop from Q30
df_c = df_c.drop(df_c.columns[96:], axis=1)

In [9]:
for col in df_c.columns:
    print(col)

Which of the following positions most closely reflects your job role? Please select all that apply.
Unnamed: 10
Unnamed: 11
Unnamed: 12
Unnamed: 13
What does school library staffing look like in your place of work or the school(s) you support? Please select all that apply.
Unnamed: 42
Unnamed: 43
Unnamed: 44
Unnamed: 45
Unnamed: 46
How often do you collaborate and work with the school librarian(s)?
What are the main reasons you do not work with or collaborate more often with the school librarian?
As an educator, how do you work with the school librarian? Please select all that apply.
Unnamed: 50
Unnamed: 51
Unnamed: 52
Unnamed: 53
Unnamed: 54
Unnamed: 55
Who usually initiates collaborations between yourself and the school librarian?
Unnamed: 57
In your opinion, what are the school librarian’s most important contributions to the education program of the school?
In which of the following ways do you engage with your school librarian(s) to effectively maximize their contribution to your a

In [10]:
df_c.shape

(43, 96)

In [11]:
df_c.head()

Which of the following positions most closely reflects your job role? Please select all that apply.  \
0                                  None of the above                                                    
1                                  None of the above                                                    
2                                  None of the above                                                    
3                                                NaN                                                    
4                                                NaN                                                    

                          Unnamed: 10                          Unnamed: 11  \
0  Teacher (e.g. educational faculty)  Support staff (e.g. teacher’s aide)   
1                                 NaN                                  NaN   
2                                 NaN                                  NaN   
3  Teacher (e.g. educational faculty)                                  NaN   
4  Teacher (e.g. educational faculty)                                  NaN   

                                         Unnamed: 12  \
0  Building Level Technology (e.g. tech specialis...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                        Unnamed: 13  \
0  Instructional Facilitator/ Coach   
1                               NaN   
2                               NaN   
3  Instructional Facilitator/ Coach   
4                               NaN   

  What does school library staffing look like in your place of work or the school(s) you support? Please select all that apply.  \
0                                  None of the above                                                                              
1                                                NaN                                                                              
2                                                NaN                                                                              
3                                                NaN                                                                              
4                                                NaN                                                                              

                             Unnamed: 42  \
0  Certified / licensed school librarian   
1                                    NaN   
2                                    NaN   
3  Certified / licensed school librarian   
4  Certified / licensed school librarian   

                                         Unnamed: 43  \
0  Paraprofessional or other non-instructional staff   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4  Paraprofessional or other non-instructional staff   

                     Unnamed: 44 Unnamed: 45  ...  \
0  Teacher on special assignment   Volunteer  ...   
1                            NaN         NaN  ...   
2                            NaN         NaN  ...   
3                            NaN         NaN  ...   
4                            NaN         NaN  ...   

                                        Unnamed: 122  \
0  Curate Make meaning for oneself and others by ...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                        Unnamed: 123  \
0  Explore Discover and innovate in a growth mind...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4    

In [12]:
df_c.rename(columns={df_c.columns[0]: 'q1_none', # Q1
                     df_c.columns[1]: 'q1_teacher',
                     df_c.columns[2]: 'q1_support_staff',
                     df_c.columns[3]: 'q1_technology',
                     df_c.columns[4]: 'q1_facilitator',

                     df_c.columns[5]: 'q5_none', # Q5
                     df_c.columns[6]: 'q5_certified',
                     df_c.columns[7]: 'q5_noninstructional',
                     df_c.columns[8]: 'q5_special_teacher',
                     df_c.columns[9]: 'q5_volunteer',
                     df_c.columns[10]: 'q5_not_sure',

                     df_c.columns[11]: 'q6_how_often', # Q6
                     df_c.columns[12]: 'q7_main_reason', # Q7

                     df_c.columns[13]: 'q8_provide_resources', # Q8
                     df_c.columns[14]: 'q8_support_lessons',
                     df_c.columns[15]: 'q8_plan_instructional_units',
                     df_c.columns[16]: 'q8_deliver_instruction',
                     df_c.columns[17]: 'q8_assess',
                     df_c.columns[18]: 'q8_special_events',
                     df_c.columns[19]: 'q8_others',

                     df_c.columns[20]: 'q9_initiate', # Q9
                     df_c.columns[21]: 'q9_others',

                     df_c.columns[22]: 'q10_librarians_contribution', # Q10

                     df_c.columns[23]: 'q11_none', # Q11
                     df_c.columns[24]: 'q11_content_department_meetings',
                     df_c.columns[25]: 'q11_grade_level_meetings',
                     df_c.columns[26]: 'q11_curriculum_planning',
                     df_c.columns[27]: 'q11_instruction_and_teaching',
                     df_c.columns[28]: 'q11_others'}
                     , inplace=True)

In [13]:
df_c.rename(columns={df_c.columns[29]: 'q12_no', # Q12
                     df_c.columns[30]: 'q12_yes_purchase',
                     df_c.columns[31]: 'q12_yes_from_workplace',
                     df_c.columns[32]: 'q12_yes_from_librarian',
                     df_c.columns[33]: 'q12_yes_not_own',
                     df_c.columns[34]: 'q12_others',

                     df_c.columns[35]: 'q13_no', # Q13
                     df_c.columns[36]: 'q13_yes_purchase',
                     df_c.columns[37]: 'q13_yes_from_workplace',
                     df_c.columns[38]: 'q13_yes_from_librarian',
                     df_c.columns[39]: 'q13_yes_not_own',
                     df_c.columns[40]: 'q13_others',

                     df_c.columns[41]: 'q14_how_familiar', # Q14
                     df_c.columns[42]: 'q15_rate', # Q15
                     df_c.columns[43]: 'q16_poor_rating_reason', # Q16

                     df_c.columns[44]: 'q17_valuable', # Q17
                     df_c.columns[45]: 'q17_relevant',
                     df_c.columns[46]: 'q17_practical',
                     df_c.columns[47]: 'q17_useful',
                     df_c.columns[48]: 'q17_easy_to_use',
                     df_c.columns[49]: 'q17_easy_to_understand',
                     df_c.columns[50]: 'q17_well_written',
                     df_c.columns[51]: 'q17_nicely_designed',
                     df_c.columns[52]: 'q17_well_organized',
                     df_c.columns[53]: 'q17_ADA_accessible',

                     df_c.columns[54]: 'q18_mirror_other_standards.', # Q18
                     df_c.columns[55]: 'q18_crosswalk_with_other_standards',
                     df_c.columns[56]: 'q18_align_educational_outlook'}
                     , inplace=True)

In [14]:
df_c.rename(columns={df_c.columns[57]: 'q19_unique_and_essential', # Q19
                     df_c.columns[58]: 'q19_effective',
                     df_c.columns[59]: 'q19_prepared',
                     df_c.columns[60]: 'q19_reading',
                     df_c.columns[61]: 'q19_intellectual_freedom',
                     df_c.columns[62]: 'q19_equitably_available',

                     df_c.columns[63]: 'q20_librarian_using', # Q20

                     df_c.columns[64]: 'q21_none', # Q21
                     df_c.columns[65]: 'q21_lesson_preparation_and_assessment',
                     df_c.columns[66]: 'q21_learning_opportunities_for_educators',
                     df_c.columns[67]: 'q21_collaboration_with_teacher',
                     df_c.columns[68]: 'q21_with_other_educators',
                     df_c.columns[69]: 'q21_with_external_stakeholders',
                     df_c.columns[70]: 'q21_with_parents',
                     df_c.columns[71]: 'q21_demonstrate_learner_success',
                     df_c.columns[72]: 'q21_guide_instructional_priorities',
                     df_c.columns[73]: 'q21_develop_curriculum',
                     df_c.columns[74]: 'q21_advocate_for_school_library',
                     df_c.columns[75]: 'q21_advocate_for_learners',
                     df_c.columns[76]: 'q21_increase_understanding_of_school_librarian',
                     df_c.columns[77]: 'q21_others',

                     df_c.columns[78]: 'q22_collaboration_aids', # Q22
                     df_c.columns[79]: 'q23_refer_to_standards', # Q23

                     df_c.columns[80]: 'q24_for_learners', # Q24
                     df_c.columns[81]: 'q24_for_school_librarians',
                     df_c.columns[82]: 'q24_for_school_libraries'}
                     , inplace=True)

In [15]:
df_c.rename(columns={df_c.columns[83]: 'q25_inquire', # Q25
                     df_c.columns[84]: 'q25_include',
                     df_c.columns[85]: 'q25_collaborate',
                     df_c.columns[86]: 'q25_curate',
                     df_c.columns[87]: 'q25_explore',
                     df_c.columns[88]: 'q25_engage',

                     df_c.columns[89]: 'q26_think', # Q26
                     df_c.columns[90]: 'q26_create',
                     df_c.columns[91]: 'q26_share',
                     df_c.columns[92]: 'q26_grow',

                     df_c.columns[93]: 'q27_rate_support_materials', # Q27
                     df_c.columns[94]: 'q28_poor_support_reason', # Q28
                     df_c.columns[95]: 'q29_additional_support'} # Q29
                     , inplace=True)

In [16]:
df_c = df_c.iloc[1:].reset_index(drop=True)

In [17]:
df_c.head()

q1_none                          q1_teacher q1_support_staff  \
0  None of the above                                 NaN              NaN   
1  None of the above                                 NaN              NaN   
2                NaN  Teacher (e.g. educational faculty)              NaN   
3                NaN  Teacher (e.g. educational faculty)              NaN   
4                NaN                                 NaN              NaN   

                                       q1_technology  \
0                                                NaN   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4  Building Level Technology (e.g. tech specialis...   

                     q1_facilitator q5_none  \
0                               NaN     NaN   
1                               NaN     NaN   
2  Instructional Facilitator/ Coach     NaN   
3                               NaN     NaN   
4                               NaN     NaN   

                            q5_certified  \
0                                    NaN   
1                                    NaN   
2  Certified / licensed school librarian   
3  Certified / licensed school librarian   
4  Certified / licensed school librarian   

                                 q5_noninstructional q5_special_teacher  \
0                                                NaN                NaN   
1                                                NaN                NaN   
2                                                NaN                NaN   
3  Paraprofessional or other non-instructional staff                NaN   
4                                                NaN                NaN   

  q5_volunteer  ... q25_curate q25_explore q25_engage q26_think q26_create  \
0          NaN  ...        NaN         NaN        NaN       NaN        NaN   
1          NaN  ...        NaN         NaN        NaN       NaN        NaN   
2          NaN  ...        NaN         NaN        NaN       NaN        NaN   
3          NaN  ...        NaN         NaN        NaN       NaN        NaN   
4          NaN  ...        NaN         NaN        NaN       NaN        NaN   

  q26_share q26_grow q27_rate_support_materials q28_poor_support_reason  \
0       NaN      NaN                        NaN                     NaN   
1       NaN      NaN                        NaN                     NaN   
2       NaN      NaN                        NaN                     NaN   
3       NaN      NaN                        NaN                     NaN   
4       NaN      NaN                        NaN                     NaN   

  q29_additional_support  
0                    NaN  
1                    NaN  
2                    NaN  
3                    NaN  
4                    NaN  

[5 rows x 96 columns]

In [18]:
# Drop the first column
df_c = df_c.iloc[:, 1:]

# Drop rows with all NaN values
df_c = df_c.dropna(how='all')

In [19]:
df_c.shape

(34, 95)

In [20]:
# Q17 - Replace extreme values with corresponding integers
df_c_q17 = df_c.iloc[:, [0] + list(range(43, 53))].dropna(
    subset=df_c.columns[43:53], how='all')

df_c_q17.iloc[:, 1:11] = df_c_q17.iloc[:, 1:11].replace({
        'Describes completely': 6,
        '2': 5,
        '3': 4,
        '4': 3,
        '5': 2,
        'Does not describe at all': 0
    }).astype(int)

In [21]:
# Q18 - Two-extremes rescale
df_c_q18 = df_c.iloc[:, [0] + list(range(53, 56))].dropna(
    subset=df_c.columns[53:56], how='all')

df_c_q18.iloc[:, 1:4] = df_c_q18.iloc[:, 1:4].replace({
    "Strongly agree": 2,
    "Somewhat agree": 1,
    "Somewhat disagree": -1,
    "Strongly disagree": -2
}).astype(int)

In [22]:
# Q19 - Two-extremes rescale
df_c_q19 = df_c.iloc[:, [0] + list(range(56, 62))].dropna(
    subset=df_c.columns[56:62], how='all')

df_c_q19.iloc[:, 1:7] = df_c_q19.iloc[:, 1:7].replace({
    "Strongly agree": 2,
    "Somewhat agree": 1,
    "Somewhat disagree": -1,
    "Strongly disagree": -2
}).astype(int)

In [23]:
# Q24 - Four levels
df_c_q24 = df_c.iloc[:, [0] + list(range(79, 82))].dropna(
    subset=df_c.columns[79:82], how='all')

df_c_q24.iloc[:, 1:4] = df_c_q24.iloc[:, 1:4].replace({
    "All the time": 3,
    "Sometimes": 2,
    "Rarely": 1,
    "Never": 0,
}).astype(int)

In [24]:
# Q25 - Four levels
df_c_q25 = df_c.iloc[:, [0] + list(range(82, 88))].dropna(
    subset=df_c.columns[82:88], how='all')

df_c_q25.iloc[:, 1:7] = df_c_q25.iloc[:, 1:7].replace({
    "All the time": 3,
    "Sometimes": 2,
    "Rarely": 1,
    "Never": 0,
}).astype(int)

In [25]:
# Q26 - Four levels
df_c_q26 = df_c.iloc[:, [0] + list(range(88, 92))].dropna(
    subset=df_c.columns[88:92], how='all')

df_c_q26.iloc[:, 1:5] = df_c_q26.iloc[:, 1:5].replace({
    "All the time": 3,
    "Sometimes": 2,
    "Rarely": 1,
    "Never": 0,
}).astype(int)

In [26]:
# OneHot encode
df_c.iloc[:, 0:4] = df_c.iloc[:, 0:4].notna().astype(int) # Q1
df_c.iloc[:, 4:10] = df_c.iloc[:, 4:10].notna().astype(int) # Q5
df_c.iloc[:, 12:18] = df_c.iloc[:, 12:18].notna().astype(int) # Q8
df_c.iloc[:, 22:27] = df_c.iloc[:, 22:27].notna().astype(int) # Q11
df_c.iloc[:, 28:33] = df_c.iloc[:, 28:33].notna().astype(int) # Q12
df_c.iloc[:, 34:39] = df_c.iloc[:, 34:39].notna().astype(int) # Q13
df_c.iloc[:, 63:76] = df_c.iloc[:, 63:76].notna().astype(int) # Q21

## Analysis

`q1`: Which of the following positions most closely reflects your job role? Please select all that apply.

In [27]:
q1_distribution = df_c.iloc[:, 0:4].sum()
print(q1_distribution)

q1_teacher          27
q1_support_staff     4
q1_technology        2
q1_facilitator       5
dtype: object


In [28]:
# Calculate the percentage
q1_percentage = (q1_distribution / q1_distribution.sum()) * 100
print("Percentage:\n", q1_percentage)

Percentage:
 q1_teacher          71.052632
q1_support_staff    10.526316
q1_technology        5.263158
q1_facilitator      13.157895
dtype: object


In [29]:
# Calculate the number of multiple choices
multiple_choices = (df_c.iloc[:, 0:4].sum(axis=1) > 1).sum()
print("Number of multiple choices:", multiple_choices)

# Filter rows where multiple choices were made
multiple_choices_rows = df_c[df_c.iloc[:, 0:4].sum(axis=1) > 1]
print("Choices for those selecting multiple options:")
print(multiple_choices_rows.iloc[:, 0:4])

Number of multiple choices: 4
Choices for those selecting multiple options:
   q1_teacher q1_support_staff q1_technology q1_facilitator
2           1                0             0              1
21          1                0             0              1
28          1                0             0              1
29          1                0             0              1


- There are 34 respondents at all.

- There are 4 respondents replying multiple options, and they chose 'Teacher (e.g. educational faculty)' and 'Instructional Facilitator/Coach'.

### Utilization

`q12`: Here is an image of the National School Library Standards for Learners, School Librarians, and School Libraries [book] produced by the American Association of School Librarians (AASL)? Please indicate whether you have seen this document before today or have a copy of this document. Please select all that apply.

In [30]:
df_c_q12 = df_c.iloc[:, 28:33]
counts = df_c_q12.sum()
percentages = (counts / 23) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q12_no                     8  34.782609
q12_yes_not_own            8  34.782609
q12_yes_purchase           4  17.391304
q12_yes_from_librarian     2   8.695652
q12_yes_from_workplace     1   4.347826

- One-third (8 of 23) have never seen the AASL Standards before — a notable awareness gap.

- Another one-third (8 of 23) have seen it but don’t own a copy, indicating passive exposure.

- Only a small number actually own the document: 4 purchased it, 1 received it from their workplace, and 2 from a librarian.

`q13`: Here is an image of the AASL Standards Framework for Learners [pamphlet] produced by the American Association of School Librarians (AASL)? Please indicate whether you have seen this document before today or have a copy of this document. Please select all that apply.

In [31]:
df_c_q13 = df_c.iloc[:, 34:39]
counts = df_c_q13.sum()
percentages = (counts / 23) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q13_no                    11  47.826087
q13_yes_not_own            7  30.434783
q13_yes_from_workplace     2   8.695652
q13_yes_from_librarian     2   8.695652
q13_yes_purchase           1   4.347826

- Nearly half (48%) of respondents have never seen the pamphlet — slightly higher than for the book.

- Only 5 of 23 (21.75%) have any kind of copy — suggesting the pamphlet may be less distributed or less visible.

- Awareness without ownership is again notable — 30% have seen it but don’t have it.

`q14`: How familiar are you with the content of the National School Library Standards for Learners, School Librarians, and School Libraries?

In [32]:
value_counts = df_c.iloc[:, 40].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q14_how_familiar                      
Not at all familiar      8   34.782609
Not too familiar         7   30.434783
Somewhat familiar        5   21.739130
Very familiar            3   13.043478

- Over two-thirds (65%) are not too familiar or not at all familiar — suggesting a significant awareness and engagement gap.

- Only 3 respondents (13%) feel very familiar, and 5 (22%) are somewhat familiar — a small but potentially strong base of informed collaborators.

`q20`: Does the school librarian use the National School Library Standards for Learners, School Librarians, and School Libraries in their work?

In [33]:
value_counts = df_c.iloc[:, 62].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q20_librarian_using                   
Yes                      8   57.142857
Not sure                 4   28.571429
No                       2   14.285714

- 57.14% (8 of 14) respondents said Yes — more than half observe active use of the Standards by the school librarian.

- However, 42.86% were either not sure (28.57%) or said No (14.29%) — signaling a possible lack of communication or visibility into how the standards are being used.

`q21`: In which of the following ways have you worked with the school librarian to implement the National School Library Standards for Learners, School Librarians, and School Libraries? Please select all that apply.

In [34]:
df_c_q21 = df_c.iloc[:, 63:76]
counts = df_c_q21.sum()
percentages = (counts / 8) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q21_collaboration_with_teacher                     7       87.5
q21_guide_instructional_priorities                 5       62.5
q21_with_other_educators                           4       50.0
q21_advocate_for_school_library                    4       50.0
q21_advocate_for_learners                          4       50.0
q21_lesson_preparation_and_assessment              3       37.5
q21_learning_opportunities_for_educators           2       25.0
q21_develop_curriculum                             2       25.0
q21_increase_understanding_of_school_librarian     2       25.0
q21_with_external_stakeholders                     1       12.5
q21_with_parents                                   1       12.5
q21_demonstrate_learner_success                    1       12.5
q21_none                                           0        0.0

- The most common form of collaboration was 'Encouraging collaboration with teacher colleagues' (7 out of 8).

- Over half worked on instructional priorities and advocacy, showing active strategic engagement.

- Few worked with external stakeholders or parents, suggesting that collaboration is more internally focused.

`q23`: How frequently do you use or refer to the National School Library Standards when working with your school librarian?

In [35]:
value_counts = df_c.iloc[:, 78].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q23_refer_to_standards                   
Rarely                      4        50.0
Sometimes                   3        37.5
Never                       1        12.5

- None of the 8 respondents use the Standards 'All the time', indicating infrequent personal engagement from collaborators.

- Most use them only 'Rarely' (50%) or 'Sometimes' (37.5%) — suggesting limited integration into their own practice.

- One respondent (12.5%) says they never personally use or refer to the Standards — even though their school librarian does.

`q24`: How often do you reference each of the following AASL Standards Frameworks when collaborating with your school librarian?

In [36]:
for col in df_c_q24.columns[1:]:
    print(df_c_q24[col].value_counts().sort_index())

q24_for_learners
2    3
Name: count, dtype: int64
q24_for_school_librarians
1    1
2    2
Name: count, dtype: int64
q24_for_school_libraries
1    1
2    2
Name: count, dtype: int64


- 'Sometimes' is the most frequent response across all three frameworks — none are referenced 'All the time'.

- The Framework for Learners is used more consistently (100% selected 'Sometimes').

- Slightly less consistency for the School Librarians and School Libraries frameworks — 1 respondent in each category selected 'Rarely'.

`q25`: How often do you consider each of the AASL Standards Shared Foundations when collaborating with your school librarian?

In [37]:
df_c_q25.iloc[:, 1:].astype(float).mean()

q25_inquire        2.333333
q25_include        2.000000
q25_collaborate    2.333333
q25_curate         2.666667
q25_explore        2.000000
q25_engage         2.000000
dtype: float64

In [38]:
counts = df_c_q25.iloc[:, 1:].apply(lambda col: col.value_counts().reindex([3, 0], fill_value=0))
print(counts)

   q25_inquire  q25_include  q25_collaborate  q25_curate  q25_explore  \
3            1            1                1           2            0   
0            0            0                0           0            0   

   q25_engage  
3           0  
0           0  


- 'Curate' stands out as the most frequently considered Shared Foundation (mean = 2.67, with 2 respondents selecting 'All the time').

- None of the Shared Foundations were marked as 'Never' — showing at least some consideration for all six domains.

- Most terms hover around a mean of ~2, suggesting that 'Sometimes' is the dominant frequency overall.

### Strength

`q15`: Overall, how would you rate the AASL National School Library Standards?

In [39]:
value_counts = df_c.iloc[:, 41].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q15_rate                    
Good           6        75.0
Excellent      2        25.0

- 100% of respondents gave a positive rating — either Good or Excellent — showing strong endorsement among those familiar with the Standards.

- No respondents selected Fair, Poor, or Not sure, indicating both clarity and perceived value of the Standards by these collaborators.

`q17`: Below is a list of words and phrases. Please indicate how well you think each word or phrase describes the National School Library Standards for Learners, School Librarians, and School Libraries.

In [40]:
df_c_q17.iloc[:, 1:].astype(float).mean()

q17_valuable              5.428571
q17_relevant              5.428571
q17_practical             4.857143
q17_useful                5.285714
q17_easy_to_use           4.571429
q17_easy_to_understand    5.000000
q17_well_written          5.142857
q17_nicely_designed       5.571429
q17_well_organized        5.000000
q17_ADA_accessible        4.857143
dtype: float64

In [41]:
counts = df_c_q17.iloc[:, 1:].apply(lambda col: col.value_counts().reindex([6, 2], fill_value=0))
print(counts)

   q17_valuable  q17_relevant  q17_practical  q17_useful  q17_easy_to_use  \
6             4             5              3           4                3   
2             0             0              0           0                1   

   q17_easy_to_understand  q17_well_written  q17_nicely_designed  \
6                       3                 3                    5   
2                       0                 0                    0   

   q17_well_organized  q17_ADA_accessible  
6                   4                   3  
2                   0                   0  


- The top-rated traits were nicely designed, valuable, and relevant, suggesting a strong appreciation for both aesthetics and substance.

- While most items scored above 5.0, 'Easy to use' had the lowest mean (4.57) and the only score of 2, pointing to usability as a key area for improvement.

- No respondents rated any trait at the bottom of the scale (0) — overall feedback is overwhelmingly positive.

`q26`: How often do you consider each of the AASL Standards Domains when collaborating with your school librarian?

In [42]:
df_c_q26.iloc[:, 1:].astype(float).mean()

q26_think     2.333333
q26_create    2.333333
q26_share     2.333333
q26_grow      2.000000
dtype: float64

In [43]:
counts = df_c_q26.iloc[:, 1:].apply(lambda col: col.value_counts().reindex([3, 2, 0], fill_value=0))
print(counts)

   q26_think  q26_create  q26_share  q26_grow
3          1           1          1         0
2          2           2          2         3
0          0           0          0         0


- Three domains — Think, Create, and Share — are equally referenced, with mean scores of 2.33, showing consistent 'sometimes' to 'often' engagement.

- 'Grow' scored slightly lower (2.00) and had no 'all the time' responses, suggesting it may receive less frequent or direct attention during collaboration.

- No respondents selected 'never' for any domain, indicating all four domains are acknowledged and at least occasionally considered.

`q27`: Overall, how would you rate AASL’s efforts to set and maintain up-to-date and practical standards and supporting resources?

In [44]:
value_counts = df_c.iloc[:, 92].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q27_rate_support_materials                   
Good                            5        62.5
Excellent                       2        25.0
Fair                            1        12.5

- A strong majority (87.5%) rated AASL’s efforts as either Good or Excellent — indicating solid overall satisfaction

- No one selected Poor or Not sure, suggesting collaborators have enough awareness to form a positive opinion.

- One person did rate it as Fair, which may reflect a desire for improvement in accessibility, relevance, or implementation tools.

### Challenge

`q7`: What are the main reasons you do not work with or collaborate more often with the school librarian?

In [45]:
df_c_q7 = df_c.iloc[:, 11].dropna().tolist()
df_c_q7

['The librarian is not approachable.',
 'no time, heavy course work requirements',
 'time, meeting standards, and the librarian at my school is amazing I believe the reason comes more from lack of time from my side. ',
 'Not enough time!',
 'Time that I have to cover material keeps me from collaborating more. ',
 'Have a library of Reading Recovery books that already use.',
 'She is not willing ',
 'To help find books the kids are interested in when we are teaching our units',
 'I work with her on certain relevant projects',
 'time',
 'Our SLM is only in our building 2 days per week with no flexibility on those days.  This makes it difficult to collaborate on lessons.  We do send students for book check-out.']

- Time is the dominant challenge — affecting both collaborators and librarians alike.

- A few responses reflect a strained relationship or limited engagement willingness.

- Others suggest that collaboration is situational, based on project relevance or resource needs.

### Suggestion

`q22`: What are some things that could be done to better aid your collaboration with your school librarian to implement the National School Library Standards?

In [46]:
df_c_q22 = df_c.iloc[:, 77].dropna().tolist()
df_c_q22

['Increasing technology',
 'my program parameters do not allow me to sway from my content at times for us to collaborate more',
 "Better alignment with other standards. They would be better integrated into the classroom teacher's standards so that collaboration would be easier and everyone would feel like we are working toward the same end. Some librarian standards are unique but there is a lot of overlap that could be investigated and used to form bridges"]

- Even with just three responses, a clear pattern emerges: collaborators want better alignment, more tech integration, and structural flexibility. These changes could help unlock more meaningful and consistent engagement with school librarians and the AASL Standards.

`q29`: What additional supports would have been helpful to you when you first encountered the National School Library Standards?

In [47]:
df_c_q29 = df_c.iloc[:, 94].dropna().tolist()
df_c_q29

['conversation with my librarian']

- The only one respondent suggested that simply having a conversation with their librarian would have been a helpful support when first encountering the AASL Standards, emphasizing the value of early, open communication.

### With Librarians

`q5`: What does school library staffing look like in your place of work or the school(s) you support? Please select all that apply.

In [48]:
df_c_q5 = df_c.iloc[:, 4:9]
counts = df_c_q5.sum()
percentages = (counts / 33) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q5_certified           31  93.939394
q5_noninstructional     7  21.212121
q5_none                 2   6.060606
q5_special_teacher      2   6.060606
q5_volunteer            0        0.0

- Nearly all collaborators (94%) reported working in a setting with a certified or licensed school librarian — a strong foundation for meaningful collaboration.

- A small portion also noted the presence of paraprofessionals or teachers on special assignment, suggesting some variation in staffing structures.

- No one selected 'Volunteer' or 'Not sure', indicating respondents were generally aware of their school’s library staffing setup.

`q6`: How often do you collaborate and work with the school librarian(s)?

In [49]:
value_counts = df_c.iloc[:, 10].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q6_how_often                              
On a monthly basis          11   35.483871
A couple times per year      8   25.806452
On a weekly basis            5   16.129032
On a daily basis             4   12.903226
Rarely, if ever              3    9.677419

- The most common collaboration frequency is monthly (35%), followed by a couple times per year (26%).

- Only 13% collaborate daily and 16% weekly, showing that frequent collaboration is less common.

- A small group (10%) reports rare or no collaboration, though this is relatively low.

`q8`: As an educator, how do you work with the school librarian? Please select all that apply.

In [50]:
df_c_q8 = df_c.iloc[:, 12:18]
counts = df_c_q8.sum()
percentages = (counts / 29) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q8_provide_resources           25  86.206897
q8_support_lessons             19  65.517241
q8_special_events              13  44.827586
q8_deliver_instruction         11  37.931034
q8_plan_instructional_units     9  31.034483
q8_assess                       1   3.448276

In [51]:
df_c.iloc[:, 18].dropna().tolist()

["I am a liason between the TL's and district Admin",
 'We bounce ideas off one another; we share newsletter/event information and opportunities',
 'basically ignored by school librarian ']

- Resource support is nearly universal — the foundational way school librarians assist educators.

- Nearly two-thirds also receive direct lesson support.

- Joint instruction and unit planning happen less frequently, suggesting collaboration tends to focus more on support than co-teaching or assessment.

- A few open-ended responses show a range from active liaison work to near non-engagement.

`q9`: Who usually initiates collaborations between yourself and the school librarian?

In [52]:
value_counts = df_c.iloc[:, 19].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q9_initiate                                                       
It varies; sometimes me, sometimes someone else     19   65.517241
I do                                                 7   24.137931
My school librarian                                  2    6.896552
Other (please specify)                               1    3.448276

In [53]:
df_c.iloc[:, 20].dropna().tolist()

["It's 50/50. Sometimes I do, and sometimes the librarian does"]

- The vast majority (about two-thirds) indicated that initiation varies, suggesting collaboration is often mutual or situational, rather than strictly led by one party.

- One in four respondents said they are the primary initiator, while very few cited the librarian alone.

- No one selected 'Administrator', suggesting that collaboration happens directly between educators and librarians, not via leadership channels.

`q10`: In your opinion, what are the school librarian’s most important contributions to the education program of the school?

In [54]:
df_c_q10 = df_c.iloc[:, 21].dropna().tolist()
df_c_q10

['Curriculum support with lessons.',
 'Development of Literacy Skills',
 'Tech and creative support',
 'core of the school and activities',
 'We would not be able to function without them. They are the backbone of our school.',
 'She’s smart and always willing to help.',
 'Katie Rainer is an innovator and creative problem solver. I love working with her to discuss ideas and make them come to life. ',
 'Providing students and teachers with valuable resources, helping students to understand how to research topics and find reliable sources, supporting teachers in lesson ideas and technology incorporation. ',
 '-providing resources and support for teachers and students  -collaboration with teachers  ',
 'Collaboration on lessons and instruction.',
 'Making reading fun, so students learn the joy of reading',
 'Our librarian provides formal instruction on library resources and good research practices. She knows YA literature, discusses books with our kids and makes recommendations based on t

- Resource Experts: Librarians are most commonly praised for providing high-quality resources that support both teachers and students. It is mentioned by 13 respondents.

- Instructional Partners: Many respondents highlighted collaboration in lesson planning and instruction as a key contribution.

- Reading & Literacy Advocates: Librarians foster a love of reading and build literacy skills through engaging materials and programs.

- Technology & Innovation Supporters: They assist with tech integration, troubleshooting, and creative tools, often serving as a go-to tech resource.

- Research & Inquiry Guides: Several responses emphasized their role in teaching information literacy and guiding students through research and inquiry-based learning.

`q11`: In which of the following ways do you engage with your school librarian(s) to effectively maximize their contribution to your academic program? Please select all that apply.

In [55]:
df_c_q11 = df_c.iloc[:, 22:27]
counts = df_c_q11.sum()
percentages = (counts / 29) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q11_instruction_and_teaching       16  55.172414
q11_content_department_meetings    10  34.482759
q11_curriculum_planning             9  31.034483
q11_none                            5  17.241379
q11_grade_level_meetings            5  17.241379

In [56]:
df_c.iloc[:, 27].dropna().tolist()

['encourage them to collaborate in their buildings as much as possible',
 'STEM ideas/interests, special events',
 'Professional development in disciplinary thinking',
 'We work together to have a successful reading program in my classes, which include regular library visit, use of library resources (digital and print) and encouraging a culture of reading. We brainstorm and plan together how to make this most effective. ']

- More than half of collaborators engage through direct instruction, showing strong integration of the librarian into teaching.

- Curriculum planning and meeting participation are secondary but meaningful areas of engagement.

- Open responses reveal creative, whole-school partnerships, from STEM and PD to school-wide reading culture.

`q19`: Please state how much you agree or disagree with each of the following Common Beliefs of the school library profession.

In [57]:
df_c_q19.iloc[:, 1:].astype(float).mean()

q19_unique_and_essential    2.000000
q19_effective               1.928571
q19_prepared                2.000000
q19_reading                 1.928571
q19_intellectual_freedom    2.000000
q19_equitably_available     2.000000
dtype: float64

In [58]:
counts = df_c_q19.iloc[:, 1:].apply(lambda col: col.value_counts().reindex([2, 1], fill_value=0))
print(counts)

   q19_unique_and_essential  q19_effective  q19_prepared  q19_reading  \
2                        14             13            14           13   
1                         0              1             0            1   

   q19_intellectual_freedom  q19_equitably_available  
2                        14                       14  
1                         0                        0  


- Unanimous support: Every belief received full agreement from respondents.

- Most beliefs were met with universal strong agreement, particularly those around intellectual freedom, equity, and preparation for life.

- A small number of responses (just 1 each) leaned toward somewhat agreeing with beliefs about librarians’ effectiveness and reading’s role.